code/iaas/api-logic/src/main/java/io/cattle/platform/iaas/api/machineDriver/MachineDriverUpdateActionHandler.java

```diff
@@ -0,0 +1,42 @@
+package io.cattle.platform.iaas.api.machineDriver;
+
+import io.cattle.platform.api.action.ActionHandler;
+import io.cattle.platform.core.constants.CommonStatesConstants;
+import io.cattle.platform.core.model.tables.records.MachineDriverRecord;
+import io.cattle.platform.object.ObjectManager;
+import io.github.ibuildthecloud.gdapi.exception.ClientVisibleException;
+import io.github.ibuildthecloud.gdapi.request.ApiRequest;
+import io.github.ibuildthecloud.gdapi.util.ResponseCodes;
+
+import javax.inject.Inject;
+
+import org.apache.commons.lang3.StringUtils;
+
+public class MachineDriverUpdateActionHandler implements ActionHandler{
+
+    @Inject
+    ObjectManager objectManager;
+
+    @Override
+    public Object perform(String name, Object obj, ApiRequest request) {
+        if (obj instanceof MachineDriverRecord) {
+            MachineDriverUpdateInput machineDriverUpdate = request.proxyRequestObject(MachineDriverUpdateInput.class);
+            ((MachineDriverRecord) obj).setMd5checksum(machineDriverUpdate.getmd5checksum());
+            if (StringUtils.isNotBlank(machineDriverUpdate.getUri())) {
+                ((MachineDriverRecord) obj).setName(machineDriverUpdate.getName());
+            }
+            if (StringUtils.isNotBlank(machineDriverUpdate.getUri())) {
+                ((MachineDriverRecord) obj).setUri(machineDriverUpdate.getUri());
+            }
+            ((MachineDriverRecord) obj).setState(CommonStatesConstants.INACTIVE);
+            objectManager.persist(obj);
+            return objectManager.reload(obj);
+        }
+        throw new ClientVisibleException(ResponseCodes.INTERNAL_SERVER_ERROR, "NotMachineDriver");
```

Comment:
None of the other actionHandlers bother with throwing this execption when the obj cant be cast as the expected type. I think you can drop the instanceof check and this ClientVisibleException and just let it bubble up.